environment_tools/node_utils.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+import re
+
+from environment_tools import type_utils
+
+NODE_NAME_FORMAT = '[A-Za-z0-9]*(-[A-Za-z0-9]*)*-(?P<location>[A-Za-z0-9]*)($|\..*)'
```

Comment:
Can you change the word "node" to just be "hostname"? 
I think it will be easier to understand when we just say hostname when we mean it and not refer to a "node's hostname"